### ISSUE TITLE
CB 2.0 DB21, Sync Gateway 1.5, Crach during sync

### ISSUE THREAD

**Issue by SteeveQuadra** [couchbase/couchbase-lite-net#945]
Hi, 

I wrote a simple windows console application (in c#) to monitor the replication status and progress, cause I've some isues.
Ma application crash during the initial replication, there are around 39000 documents in the bucket.

You can find the solution here : https://github.com/SteeveQuadra/SyncMonitor.wiki.git

I commited even the sync gateway config file I'm using, and a backup of my bucket named mobilotis.
Usage of this console app is in the wiki in gitub.

This is saddly a random issues. Even with try/catch I can't have no info about the crash.

Some time, the replication goes to the end without crashing.

Regards

Steeve

**Comment by borrrden** [comment]
I'll see if I can build and run the project.  I just need to run one instance of the app against the sync gateway configuration provided right?

**Comment by SteeveQuadra** [comment]
Yes, 

You simply run it like that :

SyncMonitor.exe username password blip://[IP adresse of syncgateway]:4984/mobilotis

"mobilotis" :  if you retore my bucket and use it.

I think you can open the solution with vs2017, restore the nuget packages, I simplified the code as much as possible for you to find the solution.

Thank you.

**Comment by borrrden** [comment]
I tried several times without experiencing a crash yet (using the provided data set).  Is there any output you can share?

**Comment by borrrden** [comment]
That being said I was having issues with log domain names being garbage on the native side.  This would seem to cause random crashing in the application that is not recoverable (See #943). I have found a workaround to that and it will be in db022 (although the release date of db022 keeps getting delayed for reasons beyond my control) so I think that might help.  

But beware, you will need to get a 2.0 build of SG for use with db022 since there is a replication protocol change.  1.5 does not work, at least not yet (I'm not sure what the plans regarding that are).

**Comment by SteeveQuadra** [comment]
Hi,

I didn't be able to catch even one exception to guide you, saddly. I will try again today.
Id did see the issue #943, and as you I though it could resolve this problem. 
I didn't expected the need to change sync gateway in order to work with the db22 of couchbase lite.
Will you or your colleagues make a developper build in link with the db22 ?

Thank you again.

Steeve

**Comment by borrrden** [comment]
Certainly of course if we require a specific version of Sync Gateway to work with db022 we will make that version easily available!  You won't have to compile sync gateway yourself or anything like that.

**Comment by SteeveQuadra** [comment]
Hi,

I finished to instal sync gateway 2.0 beta, and made all the changes in my Xamarin forms, c# solution to work with de db22 version of couchbase Lite.

I’m using a sync gateway installed on my computer, windows 10 fall creator update, to be sure it’s no network trouble.

When I’m lauching my app for the first time, I’ve around 40000 doc to sync from my bucket.
I added a log of the replicator exceptions I have that message :

LiteCoreException (1005): no status code received

I tried to restart the replicator, after some restart and the same exception, I’ve a crash of my app, and after that, the replicator exception message change to :

LiteCoreException (35): Invalid sequence ID in 'since': "39478:23249"

I’m so sad this db22 version did’t solve the replication problem I had since the db19, and db21 versions.

What can I do to solve this problems ?

Regards

Steeve
